core/src/storage2/collections/stash/mod.rs

```diff
@@ -344,49 +344,9 @@ where
         }
     }
 
-    /// Put the element into the stash at the next vacant position.
-    ///
-    /// Returns the stash index that the element was put into.
-    pub fn put(&mut self, new_value: T) -> Index {
-        let new_entry = Some(Entry::Occupied(new_value));
-        let new_index = if let Some(index) = self.last_vacant_index() {
-            // Put the new element to the most recent vacant index if not all entries are occupied.
-            let old_entry = self
-                .entries
-                .put_get(index, new_entry)
-                .expect("a `last_vacant_index()` must point to an occupied cell");
-            let vacant_entry = match old_entry {
-                Entry::Vacant(vacant_entry) => vacant_entry,
-                Entry::Occupied(_) => {
-                    unreachable!("`last_vacant_index()` must point to a vacant entry")
-                }
-            };
-            self.remove_vacant_entry(index, vacant_entry);
-            index
-        } else {
-            // Push the new element to the end if all entries are occupied.
-            let new_index = self.header.len_entries;
-            self.entries.put(new_index, new_entry);
-            self.header.last_vacant += 1;
-            self.header.len_entries += 1;
-            new_index
-        };
-        self.header.len += 1;
-        new_index
-    }
-
-    /// Takes the element stored at the given index if any.
-    pub fn take(&mut self, at: Index) -> Option<T> {
-        // Cases:
-        // - There are vacant entries already.
-        // - There are no vacant entries before.
-        if at >= self.len_entries() {
-            // Early return since `at` index is out of bounds.
-            return None
-        }
-        // Precompute prev and next vacant entries as we might need them later.
-        // Due to borrow checker constraints we cannot have this at a later stage.
-        let (prev, next) = if let Some(index) = self.last_vacant_index() {
+    /// Returns the previous and next vacant entry.
```

Comment:
Needs more documentation around the `at` parameter. Why is it returned as `(at, at)` if `self.last_vacant_index` is `None` etc.